src/components/ble/NotificationManager.cpp

```diff
@@ -67,14 +68,46 @@ NotificationManager::Notification NotificationManager::GetPrevious(NotificationM
     result = *(notifications.end() - 1);
   else
     result = *(currentIterator - 1);
-
+  
   if (result.id >= id)
-    return {};
+    return Notification {};
 
   result.index = (lastNotification.id - result.id) + 1;
   return result;
 }
 
+void NotificationManager::Dismiss(NotificationManager::Notification::Id id) {
+  if(empty)
+    return;
+
+  bool found = false;
+  size_t foundIndex;
+  size_t count = NbNotifications();
+  for(size_t i = 0; i < TotalNbNotifications; i++) {
+    if(notifications[i].id == id) {
+      foundIndex = i;
+      found = true;
+      break;
+    } 
+  }
+  if(!found)
+    return;
```

Comment:
use the available stlib functions like in `GetNext()` and `GetPrevious()` functions

```suggestion
  auto currentIterator = std::find_if(notifications.begin(), notifications.end(), [id](const Notification& n) {
    return n.valid && n.id == id;
  });
  if (currentIterator == notifications.end() || currentIterator->id != id)
    return;
  size_t foundIndex = std::distance(notifications.begin(), currentIterator);
  size_t count = NbNotifications();
```